packages/unmock-core/src/nock.ts

```diff
@@ -0,0 +1,98 @@
+import NodeBackend from "./backend";
+import { HTTPMethod } from "./interfaces";
+import { Service } from "./service";
+import { Schema } from "./service/interfaces";
+
+type UpdateCallback = ({
+  statusCode,
+  data,
+}: {
+  statusCode: number;
+  data: Schema;
+}) => Service | undefined;
+
+// Placeholder for poet input type, to have
+// e.g. standard object => { type: "object", properties: { ... }}, number => { type: "number", const: ... }
+type InputToPoet = any;
+
+export class DynamicServiceSpec {
+  private statusCode: number = 200; // TODO default success statuscode per verb?
+  private data: Schema = {};
+
+  constructor(private updater: UpdateCallback) {}
+
+  // TODO: Should this allow fluency for consecutive .get, .post, etc on the same service?
+  public reply(statusCode: number, data?: InputToPoet): Service | undefined;
+  public reply(data: InputToPoet): Service | undefined;
+  public reply(
+    maybeStatusCode: number | InputToPoet,
+    maybeData?: InputToPoet,
+  ): Service | undefined {
+    if (maybeData !== undefined) {
+      this.data = maybeData as Schema; // TODO: use poet to convert to JSON Schema?
+      this.statusCode = maybeStatusCode;
+    } else if (
+      typeof maybeStatusCode === "number" &&
+      maybeStatusCode >= 100 &&
+      maybeStatusCode < 599
+    ) {
+      // we assume it's a status code
+      this.statusCode = maybeStatusCode;
+    } else {
+      this.data = maybeStatusCode as Schema; // TODO: ditto
+    }
+    return this.updater({ data: this.data, statusCode: this.statusCode });
+  }
+}
+
+export const nockify = ({
+  backend,
+  baseUrl,
+  name,
+}: {
+  backend: NodeBackend;
+  baseUrl: string;
+  name?: string;
+}) => {
+  const dynFn = (method: HTTPMethod, endpoint: string) => ({
+    statusCode,
+    data,
+  }: {
+    statusCode: number;
+    data: Schema;
+  }) =>
+    backend.updateServices({
```

Comment:
Would it be rather possible to just fetch and/or update or create the service here to avoid too much of the http rest stuff go through the backend? The backend in my opinion should be something rather abstract.

Reply:
See my comment above - it essentially means exposing the service store from the backend. We could go down that path, although I'm not sure how that would look.

Reply:
The confusion in updating the packages like this is one reason why requiring a simple `unmock.register(nockService)` or `unmock.addService(...)` would have made the code a lot simpler if we could afford that.